src/box/iproto.cc

```diff
@@ -2107,12 +2128,35 @@ tx_process_select(struct cmsg *m)
 		obuf_rollback_to_svp(out, &svp);
 		goto error;
 	}
-	iproto_reply_select(out, &svp, msg->header.sync,
-			    ::schema_version, count);
+	if (reply_position) {
+		assert(packed_pos != NULL);
+		if (iproto_prepare_position(out) != 0) {
+			/* Discard the prepared select. */
+			obuf_rollback_to_svp(out, &svp);
+			goto error;
+		}
+		size_t packed_pos_size = packed_pos_end - packed_pos;
+		size_t alloc_size = mp_sizeof_strl(packed_pos_size);
+		char *ptr = (char *)obuf_alloc(out, alloc_size);
+		mp_encode_strl(ptr, packed_pos_size);
+		if (obuf_dup(out, packed_pos,
+			     packed_pos_size) != packed_pos_size) {
+			/* Discard the prepared select. */
+			obuf_rollback_to_svp(out, &svp);
+			goto error;
+		}
+		iproto_reply_select_with_position(out, &svp, msg->header.sync,
+						  ::schema_version, count);
```

Comment:
About `MP_STR`: all the fields of `struct request` instead of primitives (bools, ints) stores MsgPack objects, do did I. I think it's not really important where to strip the header, so let's leave it as it is.